internal/executor/resource_executor.go

```diff
@@ -0,0 +1,637 @@
+		default:
+			result[i] = v
+		}
+	}
+	return result
+}
+
+// deepCopyMap creates a deep copy of a map
+func deepCopyMap(m map[string]interface{}) map[string]interface{} {
+	// Use JSON marshaling for a simple deep copy
+	data, err := json.Marshal(m)
+	if err != nil {
+		// Fallback to shallow copy
+		result := make(map[string]interface{})
+		for k, v := range m {
+			result[k] = v
+		}
+		return result
+	}
+
+	var result map[string]interface{}
+	if err := json.Unmarshal(data, &result); err != nil {
+		// Fallback to shallow copy
+		result := make(map[string]interface{})
+		for k, v := range m {
+			result[k] = v
+		}
+		return result
+	}
+
+	return result
+}
+
+// renderManifestTemplates recursively renders all template strings in a manifest
+func renderManifestTemplates(data map[string]interface{}, params map[string]interface{}) (map[string]interface{}, error) {
+	result := make(map[string]interface{})
+
+	for k, v := range data {
+		renderedKey, err := renderTemplate(k, params)
+		if err != nil {
+			return nil, fmt.Errorf("failed to render key '%s': %w", k, err)
+		}
+
+		renderedValue, err := renderValue(v, params)
+		if err != nil {
+			return nil, fmt.Errorf("failed to render value for key '%s': %w", k, err)
+		}
+
+		result[renderedKey] = renderedValue
+	}
+
+	return result, nil
+}
+
+// renderValue renders a value recursively
+func renderValue(v interface{}, params map[string]interface{}) (interface{}, error) {
+	switch val := v.(type) {
+	case string:
+		return renderTemplate(val, params)
+	case map[string]interface{}:
+		return renderManifestTemplates(val, params)
+	case map[interface{}]interface{}:
+		converted := convertToStringKeyMap(val)
+		return renderManifestTemplates(converted, params)
+	case []interface{}:
+		result := make([]interface{}, len(val))
+		for i, item := range val {
+			rendered, err := renderValue(item, params)
+			if err != nil {
+				return nil, err
+			}
+			result[i] = rendered
+		}
+		return result, nil
+	default:
+		return v, nil
+	}
+}
+
+// getGenerationAnnotationValue extracts the generation annotation value from a resource
+// Returns 0 if the resource is nil, has no annotations, or the annotation cannot be parsed
+func getGenerationAnnotationValue(obj *unstructured.Unstructured) int64 {
+	if obj == nil {
+		return 0
+	}
+	annotations := obj.GetAnnotations()
+	if annotations == nil {
+		return 0
+	}
+	genStr, ok := annotations[AnnotationGeneration]
+	if !ok || genStr == "" {
+		return 0
+	}
+	// Try to parse as integer directly
+	gen, err := strconv.ParseInt(genStr, 10, 64)
+	if err != nil {
+		// Generation value is not a valid integer, return 0
+		return 0
+	}
+	return gen
+}
+
+// GetResourceAsMap converts an unstructured resource to a map for CEL evaluation
+func GetResourceAsMap(resource *unstructured.Unstructured) map[string]interface{} {
+	if resource == nil {
+		return nil
+	}
+	return resource.Object
+}
+
+// toCamelCase converts a snake_case string to camelCase.
```

Comment:
NP : will we need to handle kebab-case or are we happy just dealing with camelCase?